### PR TITLE
Add time to email subject and count to body

### PR DESCRIPTION
**Pull request by lmarburger**
There's an issue with Gmail threading alerts based on the subject which, as of now, includes the count. Adding the time of the alert should bust unwanted threading.

### CODE REVIEW

File: lib/papertrail_services/helpers/logs_helpers.rb [hunk 1]
```diff
@@ -70,8 +70,17 @@ def unindent(string)
...
 
-      def pluralize(count, singular, plural = nil)
-        "#{count || 0} " + ((count == 1 || count =~ /^1(\.0+)?$/) ? singular : (plural || singular.pluralize))
+      def pluralize(count, singular, plural = nil, options = {})
+        text = ""
+        unless options[:include_count] == false
+          text += "#{count || 0} "
+        end
+
+        text + ((count == 1 || count =~ /^1(\.0+)?$/) ? singular : (plural || singular.pluralize))
```
Comment (troy) [inline lib/papertrail_services/helpers/logs_helpers.rb]: I'd propose breaking this into an `if .. else` for readability

File: services/mail.rb [hunk 2]
```diff
@@ -15,7 +15,7 @@ def mail_message
       recipients = settings[:addresses].split(/(?:,|\s)+/).map { |a| a.strip }
       mail.to      recipients
       mail['reply-to'] = recipients.join(', ')
-      mail.subject %{[Papertrail] "#{payload[:saved_search][:name]}" search: #{pluralize(payload[:events].length, 'match')}}
+      mail.subject %{[Papertrail] "#{payload[:saved_search][:name]}" alert (at <%= alert_time %>)}
```
Comment (troy) [inline services/mail.rb]: I think I lean towards including it in the subject too, on the grounds that someone could see the subject and decide whether they care based on that. I can happily accept either variant, though. I like the `alert` wording.

File: lib/papertrail_services/helpers/logs_helpers.rb (hunk 1 above)
Comment (lmarburger) [inline lib/papertrail_services/helpers/logs_helpers.rb]: Overkill, but I extracted this logic into a `Pluralize` class in 815b830

File: services/mail.rb (hunk 2 above)
Comment (lmarburger) [inline services/mail.rb]: How's 3392ecf work for you?

> [Papertrail] "cron" alert, 5 matches (at 2015-02-09 08:45:08)

File: services/mail.rb [hunk 3]
```diff
@@ -15,7 +15,7 @@ def mail_message
       recipients = settings[:addresses].split(/(?:,|\s)+/).map { |a| a.strip }
       mail.to      recipients
       mail['reply-to'] = recipients.join(', ')
-      mail.subject %{[Papertrail] "#{payload[:saved_search][:name]}" search: #{pluralize(payload[:events].length, 'match')}}
+      mail.subject %{[Papertrail] "#{payload[:saved_search][:name]}" alert, #{Pluralize.new('match', :count => payload[:events].length)} (at #{alert_time})}
```
Comment (troy) [inline services/mail.rb]: A `:` became a `,` here

File: test/pluralize_test.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
+require File.expand_path('../helper', __FILE__)
+
+class PluralizeTest < PapertrailServices::TestCase
```
Comment (troy) [inline test/pluralize_test.rb]: Nice coverage!

**Comment by troy** [comment]
The `:` in the subject is all I have. Feel free to merge. It's been a while since this has changed, so I'd suggest watching for at least a few minutes on the off chance that there's exceptions in prod.

**Comment by lmarburger** [comment]
The comma was intentional but now looking back on it prefer the colon. I'll change it.

**Comment by lmarburger** [comment]
Email subject now looks like this: `[Papertrail] "cron" alert: 5 matches (at 2015-02-10 10:28:09)`

**Pull request merged**